shell/platform/darwin/common/framework/Source/FlutterChannelsTest.m

```diff
@@ -0,0 +1,142 @@
+  }
+  return self;
+}
+
+- (void)sendOnChannel:(NSString*)channel message:(NSData* _Nullable)message {
+  [self sendOnChannel:channel message:message binaryReply:nil];
+}
+
+- (void)sendOnChannel:(NSString*)channel
+              message:(NSData* _Nullable)message
+          binaryReply:(FlutterBinaryReply _Nullable)callback {
+  self.channel = channel;
+  self.message = message;
+}
+
+- (void)setMessageHandlerOnChannel:(NSString*)channel
+              binaryMessageHandler:(FlutterBinaryMessageHandler _Nullable)handler {
+  [self.handlers setObject:handler forKey:channel];
+}
+@end
+
+@interface FlutterChannelsTest : XCTestCase
+@end
+
+@implementation FlutterChannelsTest
+
+- (void)testMethodInvoke {
+  NSString* channelName = @"foo";
+  id binaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  id codec = OCMProtocolMock(@protocol(FlutterMethodCodec));
+  FlutterMethodChannel* channel = [[FlutterMethodChannel alloc] initWithName:channelName
+                                                             binaryMessenger:binaryMessenger
+                                                                       codec:codec];
+  XCTAssertNotNil(channel);
+  NSData* encodedMethodCall = [@"hey" dataUsingEncoding:NSUTF8StringEncoding];
+  OCMStub([codec encodeMethodCall:[OCMArg any]]).andReturn(encodedMethodCall);
+  [channel invokeMethod:@"foo" arguments:@[ @(1) ]];
+  OCMVerify([binaryMessenger sendOnChannel:channelName message:encodedMethodCall]);
+}
+
+- (void)testMethodInvokeWithReply {
+  NSString* channelName = @"foo";
+  id binaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  id codec = OCMProtocolMock(@protocol(FlutterMethodCodec));
+  FlutterMethodChannel* channel = [[FlutterMethodChannel alloc] initWithName:channelName
+                                                             binaryMessenger:binaryMessenger
+                                                                       codec:codec];
+  XCTAssertNotNil(channel);
+  NSData* encodedMethodCall = [@"hey" dataUsingEncoding:NSUTF8StringEncoding];
+  OCMStub([codec encodeMethodCall:[OCMArg any]]).andReturn(encodedMethodCall);
+  XCTestExpectation* didCallReply = [self expectationWithDescription:@"didCallReply"];
+  OCMExpect([binaryMessenger sendOnChannel:channelName
+                                   message:encodedMethodCall
+                               binaryReply:[OCMArg checkWithBlock:^BOOL(id obj) {
+                                 FlutterBinaryReply reply = obj;
+                                 reply(nil);
+                                 return YES;
+                               }]]);
+  [channel invokeMethod:@"foo"
+              arguments:@[ @(1) ]
+                 result:^(id _Nullable result) {
+                   [didCallReply fulfill];
+                   XCTAssertEqual(FlutterMethodNotImplemented, result);
+                 }];
+  OCMVerifyAll(binaryMessenger);
+  [self waitForExpectations:@[ didCallReply ] timeout:1.0];
```

Comment:
Sorry I meant `[self waitForExpectationsWithTimeout:1.0 handler:nil]` so you don't need to specify the expectations, it will wait for _all_ set-up expectations.  My bad.